Use of namespaces for ERC721 & ERC721Enumerable (#296)
OpenZeppelin/cairo-contracts: docs/ERC721.md

## Extensibility

> Following the [contracts extensibility pattern](Extensibility.md), this implementation is set up to include all ERC721 related storage and business logic under a namespace. Developers should be mindful of manually exposing the required methods from the namespace to comply with the standard interface. This is already done in the [preset contracts](#presets); however, additional functionality can be added. For instance, you could:

Reviewer: 👍

## Extensibility

> Following the [contracts extensibility pattern](Extensibility.md), this implementation is set up to include all ERC721 storage and function logic within the `ERC721_base.cairo` library. The `ERC721` must be must be imported and the the public facing function of this namespace must be inserted into an `external` method with the requisite name. This is already done in the [preset contracts](#presets); however, additional functionality can be added. For instance, you could:

Reviewer 2: ```suggestion
Following the [contracts extensibility pattern](Extensibility.md), this implementation is set up to include all ERC721 related storage and business logic under a namespace. Developers should be mindful of manually exposing the required methods from the namespace to comply with the standard interface. This is already done in the [preset contracts](#presets); however, additional functionality can be added. For instance, you could:
```

The `ERC721_Enumerable_Mintable_Burnable` preset adds enumerability of all the token ids in the contract as well as all token ids owned by each account. This allows contracts to publish its full list of NFTs and make them discoverable.

In regard to implementation, contracts should expose the following view methods:

> - `ERC721_Enumerable.totalSupply`
> - `ERC721_Enumerable.tokenByIndex`
> - `ERC721_Enumerable.tokenOfOwnerByIndex`

Reviewer 2: ```suggestion
- `ERC721_Enumerable.total_supply`
- `ERC721_Enumerable.token_by_index`
- `ERC721_Enumerable.token_of_owner_by_index`
```

> In order for the tokens to be correctly indexed, the contract should also use the following methods (which supercede some of the `ERC721_base` methods):

Reviewer 2: ```suggestion
In order for the tokens to be correctly indexed, the contract should also use the following methods (which supercede some of the base `ERC721` methods):
```

> - `ERC721_Enumerable.transferFrom`
> - `ERC721_Enumerable.safeTransferFrom`
> - `ERC721_Enumerable._mint`
> - `ERC721_Enumerable._burn`

Reviewer 2: ```suggestion
- `ERC721_Enumerable.transfer_from`
- `ERC721_Enumerable.safe_transfer_from`
- `ERC721_Enumerable._mint`
- `ERC721_Enumerable._burn`
```
